apps/traincascade/cascadeclassifier.cpp

```diff
@@ -450,6 +452,7 @@ void CvCascadeClassifier::save( const string filename, bool baseFormat )
         CvSeq* weak;
         if ( cascadeParams.featureType != CvFeatureParams::HAAR )
             CV_Error( CV_StsBadFunc, "old file format is used for Haar-like features only");
+        fs << "{:" << ICV_HAAR_TYPE_ID;
```

Comment:
It was my understanding that the `:` character was required but I just tried with my test program and it isn't. It does affect the output though - one line vs two lines.

```cpp
#include <opencv2/core/core.hpp>

using namespace cv;

int main() {
  std::string filename = "cascade.xml";
  FileStorage fs( filename, FileStorage::WRITE );
  fs << FileStorage::getDefaultObjectName(filename) << "{opencv-haar-classifier";
  fs << "}";
}
```

```xml
<?xml version="1.0"?>
<opencv_storage>
<cascade type_id="opencv-haar-classifier">
  </cascade>
</opencv_storage>

```